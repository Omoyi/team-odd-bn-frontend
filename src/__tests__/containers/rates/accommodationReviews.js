/* eslint-disable import/no-duplicates */
import React from 'react';
import { shallow } from 'enzyme';
import {
  AccommodationReviews,
  mapStateToProps,
} from '../../../views/accommodation/AccommodationReviews';
import {
  accommodationReviewsMocks,
  emptyAccommodationReviewsMocks,
  mapStateToProspReviews,
} from '../../../__mocks__/RatingMocks/BookingAccommodation';


describe('Expect to view accommodation reviews', () => {
  it('Should render specific accommodation with reviews successfully', () => {
    const wrapper = shallow(<AccommodationReviews {...accommodationReviewsMocks} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should not continue if review for specific accommodation is not available', () => {
    const wrapper = shallow(<AccommodationReviews {...emptyAccommodationReviewsMocks} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Shoud review accommodation from map state to propps', () => {
    const result = mapStateToProps(mapStateToProspReviews.state);
    expect(result).toEqual({ accommodationItem: { rates: 'accommodation items follows' } });
  });
});
