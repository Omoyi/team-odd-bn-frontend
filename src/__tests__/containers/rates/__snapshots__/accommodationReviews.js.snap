// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to view accommodation reviews Should not continue if review for specific accommodation is not available 1`] = `ShallowWrapper {}`;

exports[`Expect to view accommodation reviews Should render specific accommodation with reviews successfully 1`] = `ShallowWrapper {}`;
